### PR TITLE
Implement String#each_codepoint and String#codepoints

### LINKED ISSUE #1943: Implement String#each_codepoint
https://ruby-doc.org/core-2.5.3/String.html#method-i-each_codepoint
https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/String/codePointAt

### PR DESCRIPTION
**Pull request by Mogztter**
Probably a naive implementation but I was able to enable quite a few specs :wink:  

Resolves #1943

//cc @mojavelinux

### CODE REVIEW

File: opal/corelib/string/encoding.rb [hunk 1]
```diff
@@ -181,6 +181,23 @@ def each_byte(&block)
     self
   end
 
+  def each_codepoint(&block)
+    return enum_for :each_codepoint unless block_given?
+    %x{
+      for (var i = 0, length = self.length; i < length; i++) {
+        var code = self.codePointAt(i);
```
Comment (mojavelinux) [inline opal/corelib/string/encoding.rb]: The local variable assignment isn't necessary here.

Comment (Mogztter) [reply opal/corelib/string/encoding.rb]: you mean we should `yield` the result of `codePointAt` ?

```
#{yield `self.codePointAt(i)`}
```

Comment (mojavelinux) [reply opal/corelib/string/encoding.rb]: Exactly.

**Comment by elia** [comment]
I accidentally merged this one from an iOS  application that made it all too easy 😅. I’ll leave it merged, the outstanding change was a small one and I can fix it directly on master 👍

Thanks for the steady flow of PRs and fixes @mogztter!! 💪🙏

**Comment by Mogztter** [comment]
> I accidentally merged this one from an iOS  application that made it all too easy sweat_smile. 

:ship: :ship: :ship: 

> Thanks for the steady flow of PRs and fixes @Mogztter!! 

You're welcome :hugs: 

> I’ll leave it merged, the outstanding change was a small one and I can fix it directly on master

I'll take care of it :wink:

**Pull request merged**